### ISSUE TITLE
allow "continuing" process upon starting debug session

### ISSUE THREAD

**Issue by ags799** [derekparker/delve#952]
1. What version of Delve are you using (`dlv version`)?
```
Delve Debugger
Version: 1.0.0-rc.1
Build: $Id: b6077a6a35c48a31716aad2ad3fdf77bbdf70abd $
```

2. What version of Go are you using? (`go version`)?
```
go version go1.8.3 darwin/amd64
```

3. What operating system and processor architecture are you using?
```
macOS 10.12.6
2.8 GHz Intel Core i7
```

4. What did you do?
My project builds a few web servers in the spirit of microservices. We test these microservices via Docker Compose. Our existing Docker Compose configuration starts each service in its own container via [rerun](https://github.com/skelterjohn/rerun). Live-reloaded code is great, but we would like Delve's debug features (especially breakpoints).

We have tried adding Delve to our Docker Compose configuration so that each service is started with Delve (`dlv debug some/package --headless ...`). It works fine, but can be cumbersome: once a container is started, we must remote attach to its debugger & "continue" so that the web server process actually starts up.

The benefit of this -- being able to debug the web server's start up procedure -- is minimal. We rarely need to debug the start up procedure. Most of the time, we want to debug an HTTP handler.

So we'd like to use Delve, but to have it start a debug session *and* start the debugged process. Something like `dlv debug some/package --headless --continue ...`).

5. What did you expect to see?
It'd be great if Delve could start a debug session, start the process under debugging, and then allow remote connections to the debugger. That way, we can run `docker-compose up` to start our debugged services, issue REST calls and see our services work, and when needed attach a debugger & actually do some debugging.

6. What did you see instead?
It seems that Delve's existing capabilities require us to start the Docker containers (`docker-compose up`) and then, one-by-one, attach a remote debugger and "continue" the debug session. This is a bit cumbersome.

Just want to note that, despite my lack of experience in developing debuggers, I'd be happy to contribute for this feature. Also, @aarzilli has been involved in Gitter chat regarding this feature.

**Comment by ags799** [comment]
Also, I tried using the `--init` flag to do this, like so
```
dlv debug some/package --headless --init <(printf continue)
```
It caused issues with parsing the debugged process's command-line options:
```
Syntax error: "(" unexpected
```
@aarzilli seemed to think it was a bad idea too.

**Comment by lloiser** [comment]
The `--init` file is currently disabled in case you start dlv headless.

https://github.com/derekparker/delve/blob/f553c95eeb29be1694c12eb8769c3531bb628f5d/cmd/dlv/cmds/commands.go#L453-L455

Enabling it would allow you to even set some breakpoints before continuing.

**Comment by dlsniper** [comment]
This looks like a duplicate of #245

**Comment by nicksnyder** [comment]
I just ran into this behavior and find it to be inconvenient. I am debugging a web server and I would love to attach delve without pausing execution. I am also running in headless mode.

**Comment by whit537** [comment]
A related use case: I drop a `runtime.Breakpoint()` in a test and run `dlv test ./foo -- -test.run Bar`. I want to immediately run to the breakpoint in the test, I don't want to have to hit `c` first. Looks like the  `--init <(printf continue)` workaround works for this case, though. 👍

**Comment by aarzilli** [comment]
Closed by 9a216211d3461ab031f822c012bc27cab9758da0

**Comment by shiro** [comment]
sorry to bring up old issues, but is now possible to not wait for a client to connect in headless mode?

Like many other people I am developing with docker and some sort of hot-reloading.
How would I go about using delve without having to attach a debugger in order for the app to start, but rather attach later?

I tried all the combinations with `--init` (which obviously does not work with headless), `--accept-multiclient` (still waits for a client first) and without `--headless` (which does not open a remote server.

Building micro-services in docker and IntelliJ GoLand seems to be getting more popular and it would be great to support this kind of workflow.
